core/TypeErrorDiagnostics.cc

```diff
@@ -35,8 +35,10 @@ namespace {
 }
 } // namespace
 
-void TypeErrorDiagnostics::explainTypeMismatch(const GlobalState &gs, ErrorBuilder &e, const TypePtr &expected,
+void TypeErrorDiagnostics::explainTypeMismatch(const GlobalState &gs, ErrorBuilder &e,
+                                               const ErrorSection::Collector &collector, const TypePtr &expected,
                                                const TypePtr &got) {
+    e.addErrorSections(std::move(collector));
```

Comment:
Can you add a TODO that the rest of this function body should eventually be implemented in the collector? e.g., the attached class hint and the metatype hint are both things that should move into subtyping.cc eventually.